data/mods/gen3/moves.ts

```diff
@@ -414,7 +418,11 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 			},
 			onDamagePriority: -101,
 			onDamage(damage, target, source, effect) {
-				if (effect.effectType === 'Move' && !source.isAlly(target) && this.getCategory(effect.id) === 'Special') {
+				const weatherball = (effect.id === 'weatherball' && this.field.effectiveWeather() !== '');
+				if (
+					effect.effectType === 'Move' && !source.isAlly(target) &&
+					(this.getCategory(effect.id) === 'Special' || weatherball)
+				) {
```

Comment:
Weather Ball can be Rock type. Re-implementing all this Weather Ball logic in Counter and Mirror Coat doesn't seem like a great idea though.

Reply:
what do you suggest i do

Reply:
Pretty sure you can just check for `effect.category` and see if it's special. I'm pretty sure the only reason `effect.category` is not used in the first place is because of hidden power always being considered physical for counter/mirror coat, but I could be wrong on that.

Reply:
yep that seems to work, however i will need to hardcode hidden power i think